kythe/go/indexer/emit.go

```diff
@@ -933,6 +944,22 @@ func (e *emitter) writeRef(origin ast.Node, target *spb.VName, kind string) *spb
 		} else {
 			e.writeEdge(target, rule.VName, rule.EdgeOut)
 		}
+
+		if e.opts.emitDerivedLinkageFileGeneratesEdges() {
```

Comment:
This seems like the wrong location for this logic.  We're going to unnecessarily write duplicate edges between files because this loop is per reference.

Reply:
Took the same approach as C++: kept the logic here but record that a file is visited.